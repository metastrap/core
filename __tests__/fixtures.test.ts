import fs from 'node:fs';
import {
  pack, saveToLocation, setBaseProject, fixtures,
} from '@/fixtures';

jest.spyOn(fs, 'writeFileSync').mockImplementation(() => {});
jest.spyOn(fs, 'mkdirSync').mockImplementation(() => 'mock-dir-path');

describe('fixtures', () => {
  test('generate fixtures', async () => {
    const fixtureMap = await fixtures('next');
    expect(
      Array.from(fixtureMap.keys()),
    ).toMatchSnapshot();
  });
  test('pack map of files to zip', () => {
    const zip = pack(new Map([['file', 'content']]));
    expect(Object.keys(zip.files)).toStrictEqual(['file']);
  });

  test('save zip to location', async () => {
    const zip = pack(new Map([['file', 'content']]));
    await saveToLocation('./tmp', 'file', zip);
    expect(fs.mkdirSync).toHaveBeenCalledWith('./tmp', { recursive: true });
    expect(fs.writeFileSync).toHaveBeenCalledWith('./tmp/file.zip', expect.anything());
  });

  test('set base project', () => {
    const fullProject = new Map([
      ['file', 'content'],
      ['file2', 'content2'],
    ]);
    expect(
      Object.fromEntries(
        setBaseProject(fullProject, ['file']).entries(),
      ),
    ).toStrictEqual({
      file: 'content',
    });
  });
});
