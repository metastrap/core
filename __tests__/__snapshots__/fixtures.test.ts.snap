// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures generate fixtures 1`] = `
[
  ".gitignore",
  "app/globals.css",
  "app/layout.jsx",
  "app/page.jsx",
  "jsconfig.json",
  "next.config.js",
  "package.json",
  "postcss.config.js",
  "tailwind.config.js",
]
`;
